tests/unit/future/test_polling.py

```diff
@@ -192,3 +194,29 @@ def test_double_callback_background_thread():
     assert future.poll_count == 1
     callback.assert_called_once_with(future)
     callback2.assert_called_once_with(future)
+
+
+class PollingFutureImplWithoutRetry(PollingFutureImpl):
+    def done(self):
+        return True
+
```

Comment:
Please add an override of `_blocking_poll` without a `retry` parameter, since we know at least one subclass overrides `_blocking_poll`

https://github.com/googleapis/python-bigquery/blob/fb401bd94477323bba68cf252dd88166495daf54/google/cloud/bigquery/job.py#L3123

Reply:
Done